### ISSUE TITLE
False negative for protocol type compatibility with method that uses positional-only parameters

### ISSUE THREAD

**Issue by erictraut** [microsoft/pyright#6833]
The code below should produce a type error, but it doesn't currently.

```python
from typing import Any, Protocol

class Proto11(Protocol):
    def __call__(self, x: int, /, y: str) -> Any:
        ...

def cb11_bad1(x: int, y: str, /) -> Any:
    pass

cb11: Proto11 = cb11_bad1  # Type error: y is position-only
```

**Comment by erictraut** [comment]
This is addressed in pyright 1.1.344, which I just published. It will also be included in a future release of pylance.
